src/utils.c

```diff
@@ -45,11 +45,11 @@ static const char base64_decode[256] = {
 	        -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
         };
 
-static int charhex2bin(char c) {
+static uint8_t charhex2bin(char c) {
 	if(isdigit(c)) {
 		return c - '0';
 	} else {
-		return toupper(c) - 'A' + 10;
+		return toupper((uint8_t) c) - 'A' + 10;
```

Comment:
Ah yes, the dreaded ctype.h functions. Casting it to `uint8_t` is indeed correct, it should also be done for `isdigit()` in line 49 though, and there are some other `is*()` calls in the code that need fixing.